gap/PolygonalComplexes/animating.gd

```diff
@@ -0,0 +1,537 @@
+#! @Returns the updated print record.
+#! @Arguments surface, i, colour, printRecord
+DeclareOperation( "SetCircleColour", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsString, IsRecord] );
+#! @Arguments surface, i, printRecord
+DeclareOperation( "GetCircleColour", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup CalculateParametersOfInnerCircle
+#! @Description
+#! For the describtion of inner circles look at <Ref Sect="Section_LabelInnerCirclesAnimating"/>. This function calculates the parameters of the innerer circles (incenter, radius, angles relative to x-y-plane) and their normal vectors (angles relative to x-direction) based on the
+#! 3D-coordinates of the vertices and saves those to the print record.
+#! @Returns the updated print record.
+#! @Arguments surface, printRecord
+DeclareOperation( "CalculateParametersOfInnerCircle", [IsSimplicialSurface and IsNotEdgeRamified, IsRecord] );
+#! @EndGroup
+
+#! @Section Normales of Inner Circles
+#! @SectionLabel LabelNormalesInnerCirclesAnimating
+#!
+
+#! A normal of an inner circle is a vector (represented as a line) which goes orthogonal through the middle point of the corresponding inner circle. (For the describtion of inner circles look at <Ref Sect="Section_LabelInnerCirclesAnimating"/>.)
+#!
+#! The normal vectors of inner circles can also be (de-)activated in the familiar way.
+#! If the are active, they have the colour of the corresponding inner circle.
+#! @BeginExampleSession
+#! gap> oct := Octahedron();;
+#! gap> verticesPositions := [
+#! > [ 0, 0, Sqrt(2.) ],
+#! > [ 1, 1, 0 ],
+#! > [ 1, -1, 0 ],
+#! > [ -1, -1, 0 ],
+#! > [ -1, 1, 0 ],
+#! > [ 0, 0, -Sqrt(2.) ] ];;
+#! gap> printRecord := SetVertexCoordiantes3D(oct, verticesPositions, rec());;
+#! gap> printRecord := ActivateNormalOfInnerCircles(oct,printRecord);
+#! gap> DrawSurfaceToJavaScript(oct, "Octahedron_InnerCircle.html", printRecord);;
+#! @EndExampleSession
+#! @InsertChunk Example_OctahedronNormalesOfInnerCircle
+#! @BeginExampleSession
+#! gap> oct := Octahedron();;
+#! gap> verticesPositions := [
+#! > [ 0, 0, Sqrt(2.) ],
+#! > [ 1, 1, 0 ],
+#! > [ 1, -1, 0 ],
+#! > [ -1, -1, 0 ],
+#! > [ -1, 1, 0 ],
+#! > [ 0, 0, -Sqrt(2.) ] ];;
+#! gap> printRecord := SetVertexCoordiantes3D(oct, verticesPositions, rec());;
+#! gap> printRecord := DeactivateFaces(oct,printRecord);;
+#! gap> printRecord := ActivateNormalOfInnerCircles(oct,printRecord);
+#! gap> DrawSurfaceToJavaScript(oct, "Octahedron_InnerCircle.html", printRecord);;
+#! @EndExampleSession
+#! @InsertChunk Example_OctahedronNormalesOfInnerCircleWithoutFaces
+
+#! @BeginGroup ActivateNormalOfInnerCircles
+#! @Description
+#! For the describtion of normales of inner circles look at <Ref Sect="Section_LabelNormalesInnerCirclesAnimating"/>. Activate all normals innerer circles. If a normal innerer circle is active, then the normal innerer circle is shown in the animation.
```

Comment:
innerer -> inner (please also search for this mistake)
If a normal is active, the normal is shown in the animation.